stablehlo/dialect/Register.cpp

```diff
@@ -16,6 +16,7 @@ limitations under the License.
 
 #include "stablehlo/dialect/Register.h"
 
+#include "mlir/Dialect/SparseTensor/IR/SparseTensor.h"
```

Comment:
Should that have caused the build/tests to fail?

Reply:
We get it indirectly currently:
https://github.com/GleasonK/stablehlo/blob/main/stablehlo/dialect/CMakeLists.txt#L139

Should probably be added to the register target as well though.

Reply:
Our test suite doesn't yet support reliably detecting such issues. This is a known issue that MLIR-HLO also has, and addressing that is on our todo list. In the meanwhile, every now and then, we're getting bug reports / PRs like #456.

Reply:
You're right! I missed the fact that `StablehloRegister` depends on `StablehloOps`. In that case, we won't need to add the dependency.

Reply:
Probably better to have more precise dependencies in cmake though? Currently `StablehloOps` has a dependency on sparse, because it has an unused header include in `StablehloOps.cpp`. Should that be moved to the register target, since that is the only place depending on sparse currently?

Reply:
That's a nice catch! Probably a remnant of the times when we didn't yet have `BoundedAttrInterface` and `BoundedDialectInterface` and had to depend on the SparseTensor dialect directly. 

"Should that be moved to the register target, since that is the only place depending on sparse currently?". Overall, I think this is a good idea, but given that we don't yet have a good story for detecting missing dependencies, I'm worried that this might break other targets in StableHLO and/or MLIR-HLO which get their SparseTensor dependency transitively through `StablehloOps`.